cli/flox-rust-sdk/src/data/canonical_path.rs

```diff
@@ -33,7 +33,27 @@ impl CanonicalPath {
         Ok(Self(canonicalized))
     }
 
-    pub fn into_path_buf(self) -> PathBuf {
+    /// Create a [`CanonicalPath`] without checking if the path exists
+    /// or is canonical.
+    ///
+    /// This should only be used when the path is known to be canonical already.
+    pub fn new_unchecked(path: impl AsRef<Path>) -> Self {
+        Self(path.as_ref().to_path_buf())
+    }
+
+    /// Destruct the [`CanonicalPath`] and return the inner [`PathBuf`]
+    pub fn into_inner(self) -> PathBuf {
         self.0
     }
+
+    /// Get the parent directory of the path as a [`CanonicalPath`]
+    ///
+    /// Returns [`None`] if the path has no parent.
+    pub fn parent(inst: &CanonicalPath) -> Option<Self> {
+        inst.0.parent().map(Self::new_unchecked)
+    }
+
+    pub fn join(inst: &CanonicalPath, path: impl AsRef<Path>) -> Self {
+        Self::new_unchecked(inst.0.join(path))
+    }
```

Comment:
blocking: afaict this is unused, and it makes it easy to violate the guarantee of existence (which we rely on, but don't document? Probably need to update the doc comments on CanonicalPath)
```suggestion
```

Reply:
Jean you're right this is against the rules^^ I figured but didn't remove this.